packages/python/accern_data/accern_data.py

```diff
@@ -351,6 +415,15 @@ def split(
                 result.append(record)
         return result
 
+    def iterate_data(
+            self,
+            data: Optional[Union[pd.DataFrame, List[Dict[str, Any]]]],
```

Comment:
can you make the input parameters Generic[T] and then define what T is for each class

Reply:
```
def iterate_range(..., mode: Mode[T]) -> Iterate[T]:
```